plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/FileSystemSpoolingManager.java

```diff
@@ -147,11 +148,11 @@ public Optional<DirectLocation> directLocation(SpooledSegmentHandle handle)
         Optional<DirectLocation> directLocation;
         if (encryptionEnabled) {
             directLocation = fileSystem.encryptedPreSignedUri(storageLocation, ttl, key.orElseThrow())
-                    .map(uri -> new DirectLocation(uri.uri(), uri.headers()));
+                    .map(uri -> new DirectLocation(serialize(fileHandle), uri.uri(), uri.headers()));
```

Comment:
Output of `serialize(fileHandle)` is extremly chatty. Why do we need all that as identifer?

Reply:
can you elaborate? handle needs to be stateless and self-contained as we don't keep track of it on any of the nodes. thus it contains the location of the data, ttl, queryId. We can probably drop queryId

Reply:
I've removed QueryId from the handle - it's not required to retrieve or operate on the segment